### ISSUE TITLE
[Feature][CPU][Arm]: Add SME1 (FEAT_SME) dispatch for KleidiAI/QMX kernels

### ISSUE THREAD

**Issue by shivaylamba** [vllm-project/vllm#55129]
## Motivation

vLLM supports Arm CPU inference, and its W4A8 quantization path can use KleidiAI. However, I could not find a path that detects and uses first-generation Arm Scalable Matrix Extension (`FEAT_SME`/SME1) kernels independently of SME2.

This matters for recent Arm and Snapdragon CPUs that expose SME1 but may not expose SME2. On these systems, vLLM cannot take advantage of the available SME1 matrix acceleration through an explicit, observable dispatch path and instead relies on existing I8MM, NEON, or oneDNN/ACL paths.

I validated the hardware capability independently using llama.cpp with Qualcomm's KleidiAI QMX branch at commit [`8316de1358b5c3589120af0f3d99477e7c16b2e7`](https://github.com/qualcomm/kleidiai/commit/8316de1358b5c3589120af0f3d99477e7c16b2e7). On a Snapdragon-based Samsung Galaxy S26 Ultra, runtime instrumentation confirmed:

```text
primary q8 kernel feature SME
Qualcomm QMX Q8 SME1 kernels selected (qmx_mopa + qmx_dot)
executing Qualcomm QMX Q8 SME1 GEMM/prefill kernel
executing Qualcomm QMX Q8 SME1 GEMV/decode kernel
```

The prototype is Android/llama.cpp and is not directly transferable to vLLM, but it demonstrates that SME1 acceleration is available on the hardware.

References:

- Prototype: https://github.com/shivaylamba/qmx-gemma4-samsungs26ultra
- Qualcomm KleidiAI QMX branch: https://github.com/qualcomm/kleidiai/tree/kleidi-ai-qmx
- vLLM Arm CPU installation documentation: https://github.com/vllm-project/vllm/blob/main/docs/getting_started/installation/cpu.arm.inc.md
- Existing vLLM W4A8/KleidiAI documentation: https://docs.vllm.ai/en/stable/features/quantization/llm_compressor/int8_w4a8/

## Proposed feature

Please consider adding SME1 support to the Arm CPU backend with:

1. Runtime detection of `FEAT_SME` independently from `FEAT_SME2`.
2. Build support for SME1-capable kernels on AArch64.
3. SME1 dispatch for supported quantized linear GEMM/GEMV operations, initially W4A8 and/or W8A8.
4. Integration through the existing PyTorch/KleidiAI CPU path where possible.
5. Safe fallback to I8MM, NEON, or the current CPU backend when SME1 is unavailable.
6. A startup log or metrics interface showing which kernel implementation was selected.
7. Correctness tests and separate prefill/decode benchmarks against the current Arm CPU baseline.

This request is specifically for accelerating supported linear projection operations. Attention, softmax, KV-cache operations, and unsupported tensor types can remain on their existing CPU implementations.

## Questions for maintainers

- Would SME1 dispatch belong directly in vLLM's CPU backend, or should the kernel and dispatch support first be contributed to PyTorch/KleidiAI?
- Would the maintainers prefer an initial implementation limited to the existing Arm W4A8 KleidiAI path, or should it also cover W8A8?
- Is Qualcomm's KleidiAI QMX branch an acceptable optional kernel source, or would the relevant kernels need to land in upstream Arm KleidiAI first?

## Alternatives considered

The current alternatives are I8MM/NEON and the existing oneDNN/Arm Compute Library paths. Applications can maintain custom PyTorch or KleidiAI builds, but that requires downstream patches and does not provide standardized vLLM feature detection, dispatch, fallback, or observability.

SME2 kernels are not an alternative on hardware that exposes SME1 but not SME2.
